### PR TITLE
[dev] AB#21616 Ability to view a person’s personal data - [F.2] [adult record only]

### PR DESCRIPTION
**Pull request by fantaJinMode**

### CODE REVIEW

File: docs/src/modules/drawerDataGroups.js [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+
+export default class DrawerDataGroupsSample extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isDrawerOpen: false,
+        };
+
+        this._onDrawerToggle = this._onDrawerToggle.bind(this);
```
Comment (groberts314) [inline docs/src/modules/drawerDataGroups.js]: No more leading underscores.  Please adapt this file to the new ESLINT rules and style preferences.

Comment (groberts314) [reply docs/src/modules/drawerDataGroups.js]: Also note: no more multiple class components per file per the new ESLINT rules, so please split this into separate files.

File: src/modules/dataGroup.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
+import _ from 'lodash';
+import ClassNames from 'classnames';
+import Icon from '../elements/icon';
+import PropTypes from 'prop-types';
+import React from 'react';
+import Header from '../elements/header.js';
+import List from '../elements/list.js';
+import Utils from '../utils/utils.js'
+
+class DataGroupRow extends React.PureComponent {
```
Comment (groberts314) [inline src/modules/dataGroup.js]: Candidate for a functional style component...?
Please note that multiple class components per file is no longer permitted by the newer ESLINT configurations.

File: src/modules/dataGroup.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    header: PropTypes.string,
+    iconColor: PropTypes.string,
+    iconType: PropTypes.string,
+    iconSize: PropTypes.oneOfType([
+        PropTypes.oneOf(Utils.sizeEnums()),
+        PropTypes.number,
+    ]),
+};
+
+class DataGroup extends React.PureComponent {
```
Comment (groberts314) [inline src/modules/dataGroup.js]: Also a candidate for a functional style component?

File: src/modules/pageDetails.js [hunk 4]
```diff
@@ -15,7 +15,7 @@ const PageDetails = (props) => {
 PageDetails.propTypes = {
     bleed: PropTypes.bool,
     className: PropTypes.string,
-    color: PropTypes.oneOf([ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11 ]),
+    color: PropTypes.oneOf([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]),
```
Comment (groberts314) [inline src/modules/pageDetails.js]: Was this change dictated by ESLINT?

Comment (fantaJinMode) [reply src/modules/pageDetails.js]: Yes

File: src/modules/dataGroup.js (hunk 2 above)
Comment (fantaJinMode) [reply src/modules/dataGroup.js]: @groberts314 Thanks, I will split those on seperate files. Just waiting for other reviews and will start working on it

Comment (groberts314) [reply src/modules/dataGroup.js]: Sure thing, fair enough.  Thank you sir! =)

File: docs/src/modules/drawerDataGroups.js (hunk 1 above)
Comment (fantaJinMode) [reply docs/src/modules/drawerDataGroups.js]: Fixed.

File: src/modules/dataGroup.js (hunk 2 above)
Comment (fantaJinMode) [reply src/modules/dataGroup.js]: Fixed splitting multiple class components into separate files. Let's have a discussion over functional style component and `React.PureComponent`

Comment (groberts314) [reply src/modules/dataGroup.js]: > Let's have a discussion over functional style component and `React.PureComponent`

Yes, agreed.  @morethanfire @IlyaRadinsky ... what are your gents' thoughts on this matter?

File: src/modules/pageDataGroup.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        />
+    );
+};
+
+PageDataGroup.propTypes = {
+    bleed: PropTypes.bool,
+    className: PropTypes.string,
+    data: PropTypes.object.isRequired,
+    header: PropTypes.string,
+    rows: PropTypes.array.isRequired,
```
Comment (morethanfire) [inline src/modules/pageDataGroup.js]: Let's change this `PropType` `array to `arrayOf(shape)` and define the types for those propps as well.

File: src/modules/pageDataGroups.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+'use strict';
```
Comment (morethanfire) [inline src/modules/pageDataGroups.js]: Remove `use strict`

File: src/modules/pageDataGroups.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+import ClassNames from 'classnames';
+import React from 'react';
+
+const PageDataGroups = (props) => {
```
Comment (morethanfire) [inline src/modules/pageDataGroups.js]: Let's change this to a basic `function`.

File: docs/src/modules/pageDemo.js [hunk 8]
```diff
@@ -591,47 +591,98 @@ class PageDemo extends React.PureComponent {
...
+                                        ]}
+                                    />
+                                )}
+                            <Page.DataGroups
+                                style={{
+                                    margin: '22px 0'
+                                }}
+                            >
+                                <Page.DataGroup
+                                    bleed
```
Comment (morethanfire) [inline docs/src/modules/pageDemo.js]: Still not exactly sure what `bleed` is for, but I am letting it go for now.

File: src/modules/dataGroup.js (hunk 3 above)
Comment (fantaJinMode) [reply src/modules/dataGroup.js]: Fixed

File: src/modules/dataGroup.js [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+import PropTypes from 'prop-types';
+import React from 'react';
+import Header from '../elements/header.js';
+import List from '../elements/list.js';
+import DataGroupRow from './dataGroupRow.js';
+
+class DataGroup extends React.PureComponent {
+    constructor() {
+        super();
+    }
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/dataGroup.js]: Fixed

File: src/modules/dataGroupRow.js [hunk 10]
```diff
@@ -0,0 +1,94 @@
+import _ from 'lodash';
+import ClassNames from 'classnames';
+import Icon from '../elements/icon';
+import PropTypes from 'prop-types';
+import React from 'react';
+import Header from '../elements/header.js';
+import Utils from '../utils/utils.js';
+
+class DataGroupRow extends React.PureComponent {
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/dataGroupRow.js]: Fixed

File: src/modules/dataGroupRow.js [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+                fieldName,
+                header,
+                iconColor,
+                iconSize,
+                iconType,
+            },
+            data,
+        } = this.props;
+
+        const containerClasses = ClassNames('list--item_section', {
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/dataGroupRow.js]: Fixed

File: src/modules/dataGroupRow.js [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+            'header_icon': iconType || header,
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/dataGroupRow.js]: Fixed

File: src/modules/dataGroupRow.js [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+                            <Icon
+                                type={iconType}
+                                color={iconColor ? iconColor : 'primary'}
+                                size={iconSize ? iconSize : 16}
+                                style={{
+                                    marginRight: '11px'
+                                }}
+                            />
+                        )}
+                        <div className="list--item_body_section">
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/dataGroupRow.js]: Fixed

File: src/modules/dataGroupRow.js [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+                                    size="medium"
+                                    style={{
+                                        margin: 0,
+                                        lineHeight: '16px'
+                                    }}
+                                >
+                                    {header}
+                                </Header>
+                            )}
+                            <div className="list--item_data_fieldname font-size-xsmall">
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/dataGroupRow.js]: Fixed

File: src/modules/dataGroupRow.js [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+        const containerClasses = ClassNames('list--item_section', {
+            'header_icon': iconType || header,
+        });
+
+        let accessedData;
+
+        if (_.isString(accessor)) {
+            accessedData = _.get(data, accessor);
+        }
+        return (
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/dataGroupRow.js]: Fixed

File: src/modules/dataGroupRow.js [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+            <div className={containerClasses}>
+                {(header || iconType) ?
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/dataGroupRow.js]: Fixed

File: src/modules/dataGroupRow.js [hunk 17]
```diff
@@ -0,0 +1,94 @@
...
+                                >
+                                    {header}
+                                </Header>
+                            )}
+                            <div className="list--item_data_fieldname font-size-xsmall">
+                                {accessedData} | {fieldName}
+                            </div>
+                        </div>
+                    </React.Fragment> :
+                    <React.Fragment>
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/dataGroupRow.js]: Fixed

File: src/modules/dataGroupRow.js [hunk 18]
```diff
@@ -0,0 +1,94 @@
...
+                        <div className="list--item_data font-size-xsmall">
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/dataGroupRow.js]: Fixed

File: src/modules/dataGroupRow.js [hunk 19]
```diff
@@ -0,0 +1,94 @@
...
+                            {accessedData}
+                        </div>
+                        <Header className="list--item_fieldname" color="static" size="xxsmall" style={{ margin: 0 }}>
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/dataGroupRow.js]: Fixed

File: src/modules/dataGroupRow.js [hunk 20]
```diff
@@ -0,0 +1,94 @@
...
+                                size={iconSize ? iconSize : 16}
+                                style={{
+                                    marginRight: '11px'
+                                }}
+                            />
+                        )}
+                        <div className="list--item_body_section">
+                            {header && (
+                                <Header
+                                    weight="semibold"
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/dataGroupRow.js]: Fixed

File: src/modules/dataGroupRow.js [hunk 21]
```diff
@@ -0,0 +1,94 @@
...
+
+        if (_.isString(accessor)) {
+            accessedData = _.get(data, accessor);
+        }
+        return (
+            <div className={containerClasses}>
+                {(header || iconType) ?
+                    <React.Fragment>
+                        {iconType && (
+                            <Icon
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/dataGroupRow.js]: Fixed

File: src/modules/dataGroupRow.js [hunk 22]
```diff
@@ -0,0 +1,94 @@
...
+    row: PropTypes.object.isRequired,
+    header: PropTypes.string,
+    iconColor: PropTypes.string,
+    iconType: PropTypes.string,
+    iconSize: PropTypes.oneOfType([
+        PropTypes.oneOf(Utils.sizeEnums()),
+        PropTypes.number,
+    ]),
+};
+
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/dataGroupRow.js]: Fixed

File: src/modules/dataGroupRow.js [hunk 23]
```diff
@@ -0,0 +1,94 @@
...
+                        </Header>
+                    </React.Fragment>
+                }
+            </div>
+        );
+    }
+}
+
+DataGroupRow.propTypes = {
+    data: PropTypes.object.isRequired,
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/dataGroupRow.js]: Fixed

File: src/modules/dataGroupRow.js [hunk 24]
```diff
@@ -0,0 +1,94 @@
...
+    row: PropTypes.object.isRequired,
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/dataGroupRow.js]: Fixed

File: src/modules/drawerDataGroup.js [hunk 25]
```diff
@@ -0,0 +1,23 @@
+import DataGroup from './dataGroup.js';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const DrawerDataGroup = (props) => {
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/drawerDataGroup.js]: Fixed

File: src/modules/drawerDataGroup.js [hunk 26]
```diff
@@ -0,0 +1,23 @@
...
+};
+
+DrawerDataGroup.propTypes = {
+    bleed: PropTypes.bool,
+    className: PropTypes.string,
+    data: PropTypes.object.isRequired,
+    header: PropTypes.string,
+    rows: PropTypes.array.isRequired,
+    style: PropTypes.object,
+};
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/drawerDataGroup.js]: Fixed

File: src/modules/drawerDataGroup.js [hunk 27]
```diff
@@ -0,0 +1,23 @@
...
+            {...props}
+            moduleType="drawer"
+        />
+    );
+};
+
+DrawerDataGroup.propTypes = {
+    bleed: PropTypes.bool,
+    className: PropTypes.string,
+    data: PropTypes.object.isRequired,
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/drawerDataGroup.js]: Fixed

File: src/modules/drawerDataGroup.js [hunk 28]
```diff
@@ -0,0 +1,23 @@
...
+    header: PropTypes.string,
+    rows: PropTypes.array.isRequired,
+    style: PropTypes.object,
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/drawerDataGroup.js]: Fixed

File: src/modules/drawerDataGroups.js [hunk 29]
```diff
@@ -0,0 +1,26 @@
+import ClassNames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+class DrawerDataGroups extends React.PureComponent {
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/drawerDataGroups.js]: Fixed

File: src/modules/drawerDataGroups.js [hunk 30]
```diff
@@ -0,0 +1,26 @@
...
+                {children}
+            </div>
+        );
+    }
+}
+
+DrawerDataGroups.propTypes = {
+    className: PropTypes.string,
+    style: PropTypes.object,
+};
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/drawerDataGroups.js]: Fixed

File: src/modules/drawerDataGroups.js [hunk 31]
```diff
@@ -0,0 +1,26 @@
...
+            >
+                {children}
+            </div>
+        );
+    }
+}
+
+DrawerDataGroups.propTypes = {
+    className: PropTypes.string,
+    style: PropTypes.object,
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/drawerDataGroups.js]: Fixed

File: src/modules/pageDataGroup.js [hunk 32]
```diff
@@ -0,0 +1,23 @@
+import DataGroups from './dataGroup.js';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const PageDataGroup = (props) => {
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/pageDataGroup.js]: Fixed

File: src/modules/pageDataGroup.js [hunk 33]
```diff
@@ -0,0 +1,23 @@
...
+
+PageDataGroup.propTypes = {
+    bleed: PropTypes.bool,
+    className: PropTypes.string,
+    data: PropTypes.object.isRequired,
+    header: PropTypes.string,
+    rows: PropTypes.array.isRequired,
+    style: PropTypes.object,
+};
+
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/pageDataGroup.js]: Fixed

File: src/modules/pageDataGroup.js [hunk 34]
```diff
@@ -0,0 +1,23 @@
...
+            {...props}
+            moduleType="page"
+        />
+    );
+};
+
+PageDataGroup.propTypes = {
+    bleed: PropTypes.bool,
+    className: PropTypes.string,
+    data: PropTypes.object.isRequired,
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/pageDataGroup.js]: Fixed

File: src/modules/pageDataGroup.js [hunk 35]
```diff
@@ -0,0 +1,23 @@
...
+    header: PropTypes.string,
+    rows: PropTypes.array.isRequired,
+    style: PropTypes.object,
```
[the remark this answers is not in the document]
Comment (fantaJinMode) [reply src/modules/pageDataGroup.js]: Fixed

File: src/modules/pageDataGroup.js (hunk 5 above)
Comment (fantaJinMode) [reply src/modules/pageDataGroup.js]: Fixed

File: src/modules/pageDataGroups.js (hunk 6 above)
Comment (fantaJinMode) [reply src/modules/pageDataGroups.js]: Fixed

File: src/modules/pageDataGroups.js (hunk 7 above)
Comment (fantaJinMode) [reply src/modules/pageDataGroups.js]: Fixed

File: src/modules/dataGroup.js [hunk 36]
```diff
@@ -0,0 +1,70 @@
...
+    bleed: true,
+    className: undefined,
+    moduleType: undefined,
+    style: {},
+};
+
+DataGroup.propTypes = {
+    bleed: PropTypes.bool,
+    className: PropTypes.string,
+    data: PropTypes.object.isRequired,
```
Comment (morethanfire) [inline src/modules/dataGroup.js]: Change `object` to `shape`.

File: src/modules/dataGroup.js [hunk 37]
```diff
@@ -0,0 +1,70 @@
...
+    header: PropTypes.string,
+    moduleType: PropTypes.string,
+    rows: PropTypes.array.isRequired,
+    style: PropTypes.object,
```
Comment (morethanfire) [inline src/modules/dataGroup.js]: Change `object` to `shape`.

File: src/modules/dataGroup.js (hunk 36 above)
Comment (fantaJinMode) [reply src/modules/dataGroup.js]: Fixed

File: src/modules/dataGroup.js (hunk 37 above)
Comment (fantaJinMode) [reply src/modules/dataGroup.js]: Fixed

**Pull request merged**